src/core/ddsi/src/ddsi_deadline.c

```diff
@@ -105,8 +107,23 @@ void ddsi_deadline_renew_instance_real (struct ddsi_deadline_adm *deadline_adm,
      will still be triggered, but has no effect on this instance because in
      the callback the deadline (which will be the updated value) will be
      checked for expiry */
+  ddsrt_mtime_t updatetime = ddsrt_time_monotonic();
+  int64_t deadlines_missed = (updatetime.v - elem->t_last_trigger.v)/deadline_adm->dur;
+  if (deadlines_missed > 0)
+  {
+    if ((deadlines_missed + elem->deadlines_missed) > UINT32_MAX)
```

Comment:
I like the care you put in to deal with the edge cases like a 1ns deadline that result in easily overflowing the 32-bit count, but I think the way to do is so to simply say that it saturates (so no `assert`) and I personally would probably rewrite the bit to:
```
    if (deadlines_missed > UINT32_MAX || elem->deadlines_missed > UINT32_MAX - (uint32_t) deadlines_missed)
      elem->deadlines_missed = UINT32_MAX;
    else
      elem->deadlines_missed += (uint32_t) deadlines_missed;
```
but I can't say I am convinced of the wisdom of that rewrite. Yes, it'll be better on a 32-bit machine, but it will probably be worse on a 64-bit machine because of the `||`.